### ISSUE TITLE
{} + [] // -> 0  | explanation is misleading/incorrect

### ISSUE THREAD

**Issue by EmNudge** [denysdovhan/wtfjs#116]
explanation behind that says it's just coercion of one type plus another.
What is happening there is actually a code block and a unary + which coerces the array into 0.
({} + []) would get the same as ([] + {}) understandably.

**Comment by shapkarin** [comment]
Array actually not a `0` it's an empty  string `''`
That operation is equal to
```
({}).toString() + [].toString()
[].toString() + ({}).toString()
```

**Comment by libook** [comment]
```javascript
+[] // -> 0
0===+[] // -> true

{}+[] // -> 0
0==={}+[] // -> false 😕WTF??

a={}+[] // -> '[object Object]' 😕WTF??
(()=>{return {}+[]})() // -> '[object Object]'
(()=>({}+[]))() // -> '[object Object]'
```
(╯‵□′)╯︵┻━┻

**Comment by EmNudge** [comment]
Yes, because {}+[] is interpreted as 
`
{
  // a code block here
}
+[] // -> 0
`
although `0==={}+[]` is interpreted as `0 === ({} + [])`. The parens force the JS engine to recognize `{}` as an object rather than a code block.

**Comment by denysdovhan** [comment]
@EmNudge could you open a PR with a fix?

**Comment by denysdovhan** [comment]
Fixed by #150
